lib/internal/crypto/pbkdf2.js

```diff
@@ -0,0 +1,58 @@
+'use strict';
+
+const errors = require('internal/errors');
+const {
+  getDefaultEncoding,
+  toBuf
+} = require('internal/crypto/util');
+const {
+  PBKDF2
+} = process.binding('crypto');
+
+function pbkdf2(password, salt, iterations, keylen, digest, callback) {
+  if (typeof digest === 'function') {
+    callback = digest;
+    digest = undefined;
+  }
+
+  if (typeof callback !== 'function')
+    throw new errors.TypeError('ERR_INVALID_CALLBACK');
+
+  return _pbkdf2(password, salt, iterations, keylen, digest, callback);
+}
+
+function pbkdf2Sync(password, salt, iterations, keylen, digest) {
+  return _pbkdf2(password, salt, iterations, keylen, digest);
+}
+
+function _pbkdf2(password, salt, iterations, keylen, digest, callback) {
+
+  if (typeof digest !== 'string')
```

Comment:
If I'm reading the C++ code correctly, with the previous undefined check, if a digest that is neither `undefined` nor a string is passed, a default value would be used. That includes `null`. Is this something that we want to break?

https://github.com/nodejs/node/blob/bf1ca8fb2e4b92274cfc9cba837e75f7eee61b4f/src/node_crypto.cc#L5392-L5403

Reply:
you're right. this should allow `digest` to be `null` so it will default. Will fix.